R/linelist_merge.R

```diff
@@ -0,0 +1,416 @@
+      names(recode_sheet)<-paste(names(recode_sheet),variable_to, sep= ".")
+
+      require(data.table)
+      recode_sheet<-data.table(recode_sheet)
+      #merge all variables inot one column with a common key
+      recode_sheet[, paste(what_to_match,"", sep = "") :=
+                 lapply(what_to_match, function(wtm) Reduce(function(x,y) paste(x,y, sep = ","),
+                                                          .SD[, grep(wtm, names(recode_sheet)), with = F]))]
+
+      #remove the . which was a place holder for a space
+      recode_sheet <- data.frame(lapply(recode_sheet, function(x) {gsub(".", " ", x, fixed = T) }), stringsAsFactors=F)
+      #replace the NA values which have been coded as text NA after reduce
+      recode_sheet<-data.frame(lapply(recode_sheet, function(x) {gsub("NA", "", x, perl = T) }), stringsAsFactors=F)
+      #replace everything that has a more than one comma or singula commas with no characters
+      recode_sheet<-data.frame(lapply(recode_sheet, function(x) {gsub("^,*|(?<=,),|,*$", "", x, perl=T) }), stringsAsFactors=F)
+      #replace blanks with NA
+      recode_sheet[recode_sheet==""]<-NA
+      #select newly recoded columns
+      recode_sheet$id<-rownames(recode_sheet) #create an id variable for when merging back to og sheet occurs
+      recode_sheet<- recode_sheet[ ,which(names(recode_sheet) %in% c(what_to_match,"id"))]
+
+      #remove the columns that required recoding from original sheet
+      output_sheet<-og_sheet[ , -which(names(og_sheet) %in% recode_vars)]
+
+
+      #match old variable names with dictionary for new variable names
+      names(output_sheet) <- with(var_dict_country,new_var[match(names(output_sheet),old_var_r)])
+      #cbind with newly recoded variables
+      output_sheet$id<-rownames(output_sheet)#create an id variable for when merging back to og sheet occurs
+      output_sheet<-merge(output_sheet,recode_sheet,by="id") %>% select(-c(id))
+
+      # if there are no variables that needed recoding just match old variable names with dictionary for new variable names
+      }else{
+        output_sheet<- og_sheet
+        #names(output_sheet) <- gsub(x= names(output_sheet),pattern ="\r\n",replacement = "", fixed = T)    #### maybe unneccesary so comment out
+        #names(output_sheet) <-gsub(x= names(output_sheet),pattern ="\\s+", replacement =" ")         #### maybe unneccesary so comment out
+         names(output_sheet) <- with(var_dict_country,new_var[match(names(output_sheet),old_var_r)])
+      }
+      }
+
+      #handle all date variables
+      require(anytime)
+    #handle dates that are numeric. All character dates -> NA.
+    # Convert to numeric and change date format using origin of excel
+    # Convert to character date for final merge
+    output_sheet$id<-rownames(output_sheet)
+
+    datesnumeric<-output_sheet %>% select(contains("Date", ignore.case = T),id) %>% mutate_all(.funs=gsub,pattern=".",replacement = "/", fixed=T, perl = T) %>%
```

Comment:
pipes and change to across - but also it looks like anytime should be able to deal with these without having to do all the extra recoding no?